shimtest/mockstub.go

```diff
@@ -400,9 +403,33 @@ func (stub *MockStub) GetCreator() ([]byte, error) {
 	return stub.Creator, nil
 }
 
+// SetTransient to set TransientMap
+func (stub *MockStub) SetTransient(tMap map[string][]byte) error {
+	if stub.signedProposal == nil {
+		return fmt.Errorf("signedProposal is not init")
+	}
+	payload := &pb.ChaincodeProposalPayload{
+		TransientMap: tMap,
+	}
+	payloadByte, err := marshalOrError(payload)
```

Comment:
Please see comment on `marshalOrError`; this isn't reducing complexity or adding value.